backend/src/main/java/com/github/redstylzz/backend/model/MongoUser.java

```diff
@@ -40,6 +44,16 @@ public static MongoUser dtoToUser(LoginDTO dto) {
                 .build();
     }
 
+    @Transient
+    public static MongoUser getUser(UsernamePasswordAuthenticationToken principal, Log log) throws ResponseStatusException {
+        try {
+            return (MongoUser) principal.getPrincipal();
+        } catch (Exception e) {
+            log.warn("No user found");
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "No user found");
+        }
+    }
+
```

Comment:
I dont like extracting this to the model class. I would keep this in the controller.